nemo/lightning/pytorch/trainer.py

```diff
@@ -13,3 +28,120 @@ def io_init(self, **kwargs) -> fdl.Config[Self]:
+            ValueError: If log_local_rank_0_only and log_global_rank_0_only are both True
+        """
+        local_rank = int(os.environ.get("LOCAL_RANK", 0))
+        global_rank = self.node_rank * self.num_devices + local_rank
+        logging.rank = global_rank
+
+        if explicit_log_dir:  # If explicit log_dir was passed, short circuit
+            return check_explicit_log_dir(self, explicit_log_dir, exp_dir, name, version)
+
+        # Default exp_dir to ./nemo_experiments if None was passed
+        _exp_dir = exp_dir
+        if exp_dir is None:
+            _exp_dir = str(Path.cwd() / 'nemo_experiments')
+
+        if not name:
+            name = "default"
+
+        if self.logger is not None:
+            if update_logger_directory:
+                logging.warning(
+                    f'"update_logger_directory" is True. Overwriting logger "save_dir" to {exp_dir} and "name" to {name}'
+                )
+                self.logger._root_dir = exp_dir
+                self.logger._name = name
+
+        version = version or os.environ.get(NEMO_ENV_VARNAME_VERSION, None)
+        if is_global_rank_zero():
+            if use_datetime_version:
+                version = time.strftime('%Y-%m-%d_%H-%M-%S')
+        if resume.resume_if_exists:
+            logging.warning(
+                "No version folders would be created under the log folder as 'resume_if_exists' is enabled."
+            )
+            version = None
+        if version:
+            if is_global_rank_zero():
+                os.environ[NEMO_ENV_VARNAME_VERSION] = version
+
+        log_dir = Path(_exp_dir) / Path(str(name)) / Path("" if version is None else str(version))
+        # update app_state with log_dir, exp_dir, etc
+        app_state = AppState()
+        app_state.log_dir = log_dir
+        app_state.exp_dir = exp_dir
+        app_state.name = name
+        app_state.version = version
+
+        # Create the logging directory if it does not exist
+        os.makedirs(log_dir, exist_ok=True)  # Cannot limit creation to global zero as all ranks write to own log file
+        logging.info(f'Experiments will be logged at {log_dir}')
+        self._default_root_dir = log_dir
+
+        ckpt_path = resume.nemo_path()
+        if ckpt_path:
+            self.ckpt_path = ckpt_path
+
+        ## TODO: assert that we only have one checkpoint callback?
+        for callback in self.callbacks:
+            if isinstance(callback, PTLModelCheckpoint):
+                ## TODO: make configurable
+                callback.dirpath = Path(log_dir / "checkpoints")  # app_state.exp_dir
+                if callback.filename is None:
+                    callback.filename = f'{name}--{{{callback.monitor}:.4f}}-{{epoch}}'
+                if callback.prefix is None:
+                    callback.prefix = name
+                ModelCheckpoint.CHECKPOINT_NAME_LAST = callback.filename + '-last'
```

Comment:
Shouldn't we use our `ModelCheckpoint` by default if the users hasn't explicitly passed a ModelCheckpoint class into the Trainer?